### PR TITLE
Persist and visualize conflicts in modeling assessments

### PR DESCRIPTION
**Pull request by jmaximilian2**
### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors.
- [x] I run `yarn lint`: the project builds without code style warnings.
- [x] I updated the documentation and models.
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~I added (end-to-end) test cases for the new functionality.~

### Motivation and Context

Integrates conflict resolution view into assessment workflow and persists upcoming conflicts in database

### Description

Cherry picked version of the persist conflict PR (https://github.com/ls1intum/ArTEMiS/pull/376).

- Enables the tutor to review upcoming conflicts besides each other and decide wether to apply the already existing assessment of an element to his own assessment or escalate the conflict.
- Updated Gatling files to syntax of version 3.1
- Added gatling ConflictSimulation for creating new course with exercise with 2 submissions and 1 assessment
- Added and extended conflict related integration tests

### Steps for Testing

1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Produce Conflict
4. Use Resolve Conflicts button to enter new view

### Screenshots

Bildschirmfoto 2019-04-30 um 13 26 29
Bildschirmfoto 2019-04-30 um 13 26 13

### CODE REVIEW

**Comment by thilo-behnke** [comment]
From a code-wise perspective it looks good to me now.
Only the documentation could still be improved a bit:
E.g. src/main/webapp/app/modeling-assessment-editor/modeling-assessment-conflict/modeling-assessment-conflict.component.ts has a couple of not self-explanatory methods (e.g. `onSubmit` does a lot of things, `updateFeedbackInMergedFeedback` and `updateCurrentState` are unclear to me

File: src/main/webapp/app/modeling-assessment-editor/modeling-assessment-editor.component.ts [hunk 1]
```diff
@@ -206,23 +205,23 @@ export class ModelingAssessmentEditorComponent implements OnInit, OnDestroy {
     onSubmitAssessment() {
         // TODO: we should warn the tutor if not all model elements have been assessed, and ask him to confirm that he really wants to submit the assessment
         // in case he says no, we should potentially highlight the elements that are not yet assessed
-        this.modelingAssessmentService.saveAssessment(this.feedback, this.submission.id, true, this.ignoreConflicts).subscribe(
+        this.modelingAssessmentService.saveAssessment(this.feedback, this.submission.id, true, false).subscribe(
```
Comment (krusche) [inline src/main/webapp/app/modeling-assessment-editor/modeling-assessment-editor.component.ts]:
```suggestion
        this.modelingAssessmentService.saveAssessment(this.feedback, this.submission.id, true, true).subscribe(
```

File: src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java [hunk 2]
```diff
@@ -193,20 +204,20 @@ public void addAssessment(long exerciseId, long submissionId, List<Feedback> mod
         }
     }
 
-    public List<Conflict> getConflicts(ModelingSubmission modelingSubmission, long exerciseId, Result result, List<Feedback> modelingAssessment) {
+    public List<ModelAssessmentConflict> getConflicts(ModelingSubmission modelingSubmission, long exerciseId, Result result, List<Feedback> modelingAssessment) {
```
[the remark this answers is not in the document]
Comment (jmaximilian2) [reply src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java]: Totally agree with you will be changed in the follow up branch

File: src/main/webapp/app/modeling-assessment/modeling-assessment.component.ts [hunk 3]
```diff
@@ -79,12 +82,26 @@ export class ModelingAssessmentComponent implements AfterViewInit, OnDestroy, On
         if (changes.feedbacks && changes.feedbacks.currentValue && this.model) {
             this.feedbacks = changes.feedbacks.currentValue;
             this.handleFeedback();
+            this.updateElementFeedbackMapping(this.feedbacks);
+            this.updateApollonAssessments(this.feedbacks);
+            this.calculateTotalScore();
```
[the remark this answers is not in the document]
Comment (jmaximilian2) [reply src/main/webapp/app/modeling-assessment/modeling-assessment.component.ts]: Thanks, fixed  :+1:

File: src/main/webapp/app/modeling-assessment-editor/modeling-assessment-conflict/modeling-assessment-conflict.component.html [hunk 4]
```diff
@@ -1,40 +1,60 @@
...
-        ></jhi-modeling-assessment>
+        <div class="conflicting-assessment">
+            <jhi-modeling-assessment
+                *ngIf="currentConflict"
+                [diagramType]="modelingExercise.diagramType"
+                [model]="conflictingModel"
+                [feedbacks]="conflictingResult.result.feedbacks"
+                title="{{ 'modelingAssessmentEditor.conflictingAssessmentTitle' | translate: { count: currentConflict.resultsInConflict.length } }} {{
+                    conflictingResult.result.assessor.firstName
+                }} {{ conflictingResult.result.assessor.lastName }}"
```
[the remark this answers is not in the document]
Comment (jmaximilian2) [reply src/main/webapp/app/modeling-assessment-editor/modeling-assessment-conflict/modeling-assessment-conflict.component.html]: will be deprecated by the follow up branch conflict-notification

**Pull request merged**